Research/Research/RSDFormDataType.swift

```diff
@@ -55,6 +55,10 @@ public enum RSDFormDataType {
     /// Default range is for an adult.
     case measurement(MeasurementType, MeasurementRange)
     
+    /// A "detail" form data type is a data type that represents an input field where the data entry uses a
+    /// detail that displays information using one or more input fields. The default base type is `.codable`.
```

Comment:
should this be 'detail view' (or 'detail UI' or similar if you want to be more generic)? or am I misunderstanding the gist of this doc comment?

Reply:
So... this is where things get kinda confusing. Basically, the "detail" **data type** is any codable object that can be encoded/decoded into an answer. Since this might be a single data type, the class that does this, forwards the data type from its wrapped input fields, if applicable.

Really, it's a work-around to allow us to support a single "type" keyword field for Android deserialization and casting. In Swift, it's fairly straight-forward to use two keys, one that defines the class type and another that defines the data type, but apparently that's a lot of yoga on Android.